txnbuild/transaction.go

```diff
@@ -955,6 +956,11 @@ func ReadChallengeTx(challengeTx, serverAccountID, network, homeDomain string) (
 	if op.SourceAccount == nil {
 		return tx, clientAccountID, errors.New("operation should have a source account")
 	}
+	if strings.Split(op.Name, " ")[0] != homeDomain {
+		return tx, clientAccountID, errors.Errorf("operation key should contain homeDomain (homeDomain=\"%s\", key=\"%s\")",
+			homeDomain, op.Name)
```

Comment:
💯  for including the details of the key and home domain for better debugging.

I suggest reordering the parameters at the end so they are in order of the concepts introduced in the string. A reader reads the sentence left to right and it is more intuitive to compare the key and home domain in the same order as the sentence that introduced them.

You can also use the `%q` notation to render a string as a Go literal which for a string means in quotes with any quotes within escaped so you don't have to do that manually.
```suggestion
		return tx, clientAccountID, errors.Errorf("operation key should contain homeDomain (key=%q, homeDomain=%q)", homeDomain, op.Name)
```